mlflow/tensorflow.py

```diff
@@ -1044,40 +1039,27 @@ def fit_generator(self, *args, **kwargs):
 
             return result
 
-    @gorilla.patch(EventFileWriter)
     def add_event(self, event):
         _log_event(event)
         original = gorilla.get_original_attribute(EventFileWriter, "add_event")
         return original(self, event)
 
-    @gorilla.patch(FileWriter)
     def add_summary(self, *args, **kwargs):
         original = gorilla.get_original_attribute(FileWriter, "add_summary")
         result = original(self, *args, **kwargs)
         _flush_queue()
         return result
 
-    settings = gorilla.Settings(allow_hit=True, store_hit=True)
     patches = [
-        gorilla.Patch(EventFileWriter, "add_event", add_event, settings=settings),
-        gorilla.Patch(EventFileWriterV2, "add_event", add_event, settings=settings),
-        gorilla.Patch(tensorflow.estimator.Estimator, "train", train, settings=settings),
-        gorilla.Patch(tensorflow.keras.Model, "fit", fit, settings=settings),
-        gorilla.Patch(tensorflow.keras.Model, "fit_generator", fit_generator, settings=settings),
-        gorilla.Patch(
-            tensorflow.estimator.Estimator,
-            "export_saved_model",
-            export_saved_model,
-            settings=settings,
-        ),
-        gorilla.Patch(
-            tensorflow.estimator.Estimator,
-            "export_savedmodel",
-            export_savedmodel,
-            settings=settings,
-        ),
-        gorilla.Patch(FileWriter, "add_summary", add_summary, settings=settings),
+        (EventFileWriter, "add_event", add_event),
+        (EventFileWriterV2, "add_event", add_event),
+        (tensorflow.estimator.Estimator, "train", train),
+        (tensorflow.keras.Model, "fit", fit),
+        (tensorflow.keras.Model, "fit_generator", fit_generator),
+        (tensorflow.estimator.Estimator, "export_saved_model", export_saved_model),
+        (tensorflow.estimator.Estimator, "export_savedmodel", export_savedmodel),
+        (FileWriter, "add_summary", add_summary),
```

Comment:
@dbczumar Should we make the `name` argument of `wrap_patch` optional so we can omit it when a patch function has the same name as a target attribute?

```python
def wrap_patch(dest, patch, name=None, ...):
    name = patch.__name__ if name is None else name
    ...
```

Reply:
I don't think that optimization is strictly necessary. `gorilla` doesn't do it, for example.

Reply:
makes sense